js/main.js

```diff
@@ -0,0 +1,219 @@
+let weatherInfo = {
+    loc: null,
+    K: null,
+    F: null,
+    C: null,
+    con: null,
+    icon: null, 
+    mCon: null,
+};
```

Comment:
Sometimes it's better to start the defaults with typehinting. Meaning throwing in a falsy but similar type variables. `''` for empty strings, 0 for numbers, etc.

Reply:
switched to typehinting